### PR TITLE
 Fix BonusListID for profession created items 

### PR DESCRIPTION
**Pull request by Palabola**
**Changes proposed:**

-  
-  
-  

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

### CODE REVIEW

File: src/server/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -1442,7 +1442,13 @@ void Spell::EffectCreateItem(SpellEffIndex effIndex)
     if (effectHandleMode != SPELL_EFFECT_HANDLE_HIT_TARGET)
         return;
 
-    DoCreateItem(effIndex, effectInfo->ItemType);
+    std::set<uint32> bonusListIDs = sDB2Manager.GetItemBonusTree(effectInfo->ItemType, 13); // 13 is guessed tradeskill bonus mod
+
+	std::vector<int32> bonusList;
```
Comment (Shauren) [inline src/server/game/Spells/SpellEffects.cpp]: tabs

**Review by Shauren — CHANGES_REQUESTED** [review]

File: src/server/game/Spells/SpellEffects.cpp (hunk 1 above)
Comment (soulfrost) [reply src/server/game/Spells/SpellEffects.cpp]: itemBonusTreeMod must not be hardcoded there, check it more

**Comment by Aokromes** [comment]
TrinityCore/src/server/game/Spells/SpellEffects.cpp:1451:50: fatal error: no viable conversion from 'std::vector<int32>' (aka 'vector<int>') to 'uint8' (aka 'unsigned char')
    DoCreateItem(effIndex, effectInfo->ItemType, bonusList);

File: src/server/game/Spells/SpellEffects.cpp (hunk 1 above)
Comment (Aokromes) [reply src/server/game/Spells/SpellEffects.cpp]: tabs2spaces

**Comment by jackpoz** [comment]
Closing this PR as it seems abandoned. Feel free to ask here to reopen it if anyone would like to work on it.

**Pull request closed without merging**